README.md

```diff
@@ -1110,6 +1110,175 @@ Generally, either the default implementation or the override closures should pro
 
 > It is important to keep in mind that the `subdelegates` are initialized and destroyed in the default implementation. Be careful when subclassing to not introduce memory leaks.
 
+### Adapting and Retrying Requests
```

Comment:
In general, do we want any discussion of when to use an adapter vs. when to use a router?